src/AccessToken.php

```diff
@@ -103,57 +101,129 @@ public function verify($token, array $options = [])
         $certsLocation = isset($options['certsLocation'])
             ? $options['certsLocation']
             : self::FEDERATED_SIGNON_CERT_URL;
-
-        unset($options['audience'], $options['certsLocation']);
+        $cacheKey = isset($options['cacheKey'])
+            ? $options['cacheKey']
+            : $this->getCacheKeyFromCertLocation($certsLocation);
 
         // Check signature against each available cert.
         // allow the loop to complete unless a known bad result is encountered.
-        $certs = $this->getFederatedSignOnCerts($certsLocation, $options);
+        $certs = $this->getCerts($certsLocation, $cacheKey, $options);
+        $alg = $this->determineAlg($certs);
+
+        switch ($alg) {
+            case 'ES256':
+                return $this->verifyEs256($token, $certs, $audience);
+
+            case 'RS256':
+                return $this->verifyRs256($token, $certs, $audience);
+
+            default:
+                throw new \InvalidArgumentException(
+                    'unrecognized "alg" in certs, expected ES256 or RS256');
+        }
+    }
+
+    private function determineAlg($certs)
```

Comment:
would you mind documenting the private methods?

Reply:
Done.